src/ngrest/plugins/Injector.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace luya\admin\ngrest\plugins;
+
+use luya\admin\ngrest\base\Plugin;
+use luya\helpers\Html as HtmlHelper;
+
+/**
+ * Create a generic form input based on ZAA Directives.
+ *
+ * ```php
+ *   public function ngRestAttributeTypes()
+ *   {
+ *      return [
+ *          'name' => 'text',
+ *          'type' => 'text',
+ *          'value' => ['injector', 'directiveField' => 'type'],
+ *      ];
+ *   }
+ * ```
+ *
+ * @author Bennet Klarhoelter <boehsermoe@me.com>
+ * @since 1.2.3
+ */
+class Injector extends Plugin
+{
+    /**
+     * @var string Property name from the model to use as ZAA Directive.
+     */
+    public $directiveField;
```

Comment:
@boehsermoe Sorry for bringing the idea a little late, but maybe it should be `public $attribute`? As it makes it more clear it must be one of `ngRestAttributeTypes`.

Also some ideas:

1. Maybe it would be possible to check whether this attribute exists or not - as $renderContext provides RenderCrudInterface with model access.